libc/stdio/fmt.c

```diff
@@ -1073,6 +1073,9 @@ int __fmt(void *fn, void *arg, const char *format, va_list va, int *wrote) {
         }
         break;
       }
+      case 'C':
+        signbit = 63;
```

Comment:
`wchar_t` / `wint_t` are 32-bit types. Why is signbit set to 63?

Reply:
`__fmt_stoa` only considers the argument to be of type `wchar_t` if `signbit == 63`. Otherwise, it's `char16_t` if `signbit == 15` and otherwise it's `char`. I suppose the way `wchar_t` conversions in general could be changed so it uses a different value of `signbit`, but without such major changes to other parts of `fmt.c`, there isn't really any way to implement the `C` conversion specifier without having `signbit` end up with a value of `63`

Reply:
Will it say `va_arg(va, long)`? Because if that's the case, you should write a test that does `printf("%d%d%d%d%d%d%d%d%C%d\n", 0,0,0,0,0,0,0,0, L'x', 1)` to prove it won't break.

Reply:
If that breaks `%C`, then I would also expect it to break `%lc` (I'll try it, though)

Reply:
Neither break, I'll amend my commit with the corresponding test.

Reply:
(this is because the code in `case 'c':` 2 lines later unconditionally does `va_arg(va, int)`)

Reply:
I've pushed the amended commit.

Reply:
(uh, now it should actually work, sorry for that lol)